File: connectx-backend/src/main/java/com/connectx/user/dto/UserProfileUpdateDto.java
package com.connectx.user.dto;

public class UserProfileUpdateDto {

    private String username;
    private String displayName;
    private String profileImageUrl;
    private String status;
    private String profilePhotoVisibility;

    public UserProfileUpdateDto() {}

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public String getProfileImageUrl() {
        return profileImageUrl;
    }

    public void setProfileImageUrl(String profileImageUrl) {
        this.profileImageUrl = profileImageUrl;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getProfilePhotoVisibility() {
        return profilePhotoVisibility;
    }

    public void setProfilePhotoVisibility(String profilePhotoVisibility) {
        this.profilePhotoVisibility = profilePhotoVisibility;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/user/model/UserProfileUpdateDto.kt
package com.connectx.app.data.remote.user.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.user.dto.UserProfileUpdateDto exactly. profileImageUrl is
 * deliberately NOT a field -- the backend intentionally excludes it here (photo
 * changes only via the dedicated upload/remove endpoints; see the backend class's
 * javadoc for the security reasoning). No validation annotations exist on the
 * backend DTO, so all fields are optional/nullable here too.
 */
@Serializable
data class UserProfileUpdateDto(
    val username: String? = null,
    val displayName: String? = null,
    val status: String? = null,
    val profilePhotoVisibility: String? = null,
    val groupAddPrivacy: String? = null
)
